controllers/metal3.io/host_state_machine.go

```diff
@@ -428,6 +428,27 @@ func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
 }
 
 func (hsm *hostStateMachine) provisioningCancelled() bool {
+	if hsm.Host.Spec.CustomDeploy != nil && hsm.Host.Spec.CustomDeploy.Method != "" {
+		if hsm.Host.Status.Provisioning.CustomDeploy != nil && hsm.Host.Spec.CustomDeploy.Method != hsm.Host.Status.Provisioning.CustomDeploy.Method {
```

Comment:
The `Status.Provisioning.CustomDeploy.Method` is expected to be empty during provisioning. I think _in practice_ that `Status.Provisioning.CustomDeploy` will be nil also, but it seems like it would be more robust to check both:

```
if hsm.Host.Status.Provisioning.CustomDeploy != nil &&
    hsm.Host.Status.Provisioning.CustomDeploy.Method != "" &&
        hsm.Host.Spec.CustomDeploy.Method != hsm.Host.Status.Provisioning.CustomDeploy.Method {
```

Reply:
May I follow-up with this? I'd prefer to avoid another round of looking for lgtm and fighting the CI because of something so minor.

Reply:
https://github.com/metal3-io/baremetal-operator/pull/932